api/rating.py

```diff
@@ -0,0 +1,295 @@
+    'rating': 'ROUND(r.mean - 3 * r.deviation)',
+    'ranking': '@rownum:=@rownum+1'
+}
+
+TABLE1V1 = 'ladder1v1_rating r JOIN login l on r.id = l.id, (SELECT @rownum:=%(row_num)s) n'
+TABLEGLOBAL = 'global_rating r JOIN login l on r.id = l.id, (SELECT @rownum:=%(row_num)s) n'
+
+@app.route('/rating/<string:rating_type>')
+def rating_type(rating_type):
+    """
+        Lists all ranked 1v1 or global players.
+
+        **Example Request**:
+
+        **Default Values**:
+            page[number]=1
+
+            page[size]=5000
+
+        .. sourcecode:: http
+
+           GET /rating/1v1 /rating/global
+           Accept: application/vnd.api+json
+
+        **Example Response**:
+
+        .. sourcecode:: http
+
+            HTTP/1.1 200 OK
+            Vary: Accept
+            Content-Type: text/javascript
+
+            {
+              "data": [
+                {
+                  "attributes": {
+                    "deviation": 48.4808,
+                    "id": "781",
+                    "is_active": true,
+                    "login": "Zock",
+                    "mean": 2475.69,
+                    "num_games": 1285,
+                    "ranking": 1,
+                    "rating": 2330,
+                    "won_games": 946
+                  },
+                  "id": "781",
+                  "type": "ranked1v1"
+                },
+                ...
+              ]
+            }
+
+        :param page[number]: The page number being requested (EX.: /rating/1v1?page[number]=2)
+        :type page[number]: int
+        :param page[size]: The total amount of players to grab by default (EX.: /rating/1v1?page[size]=10)
+        :type page[size]: int
+        :param filter[isActive]: Whether or not to filter active players or not (true or false) (EX.: /rating/1v1?filter[is_active]=true)
+        :type filter[isActive]: boolean
+        :param filter[player]: Allows search functionality in the ranked1v1 endpoint based upon the players login name (EX.: /rating/1v1?filter[player]=Zock)
+        :type filter[player]: name
+        :param rating_type: Finds players in the 1v1 or global rating
+        :type rating_type: 1v1 OR global
+        :status 200: No error
+
+        """
+    sort_field = request.values.get('sort')
+    if sort_field:
+        raise ApiException([Error(ErrorCode.QUERY_INVALID_SORT_FIELD, sort_field)])
+
+    page = int(request.values.get('page[number]', 1))
+    page_size = int(request.values.get('page[size]', MAX_PAGE_SIZE))
+    player = request.args.get('filter[player]')
+    row_num = (page - 1) * page_size
+    select = SELECT_EXPRESSIONS
+
+    args = {'row_num': row_num}
+
+    where = ''
+    active_filter = request.values.get('filter[is_active]')
+    if active_filter:
+        where += 'is_active = ' + ('1' if active_filter.lower() == 'true' else '0') + ' AND r.numGames > 0'
+
+    if player:
+        where += " l.login LIKE %(player)s"
+        args['player'] = '%' + player + '%'
```

Comment:
We still can't filter by players, because the ranking is determined by counting the number of rows in the result. Everything else would end up in a super expensive query.

Please remove all filters and throw an explaining ApiException if somebody tries to filter.
The `is_active` where clause should always be added.